server/services/starUpgrade.js

```diff
@@ -177,7 +178,8 @@ module.exports = class StarUpgradeService extends EventEmitter {
         // Calculate how much the upgrade will cost.
         const expenseConfig = game.constants.star.infrastructureExpenseMultipliers[expenseConfigKey];
         const terraformedResources = this.starService.calculateTerraformedResources(star.naturalResources, effectiveTechs.terraforming);
```

Comment:
`this.starService.calculateTerraformedResources(star.naturalResources[economyType], effectiveTechs.terraforming);`?

Reply:
Done